File: gui_view/src/main/scala/com/goexp/galgame/gui/util/TabSelect.java
package com.goexp.galgame.gui.util;

import com.goexp.galgame.gui.view.game.HomeController;
import javafx.scene.control.Tab;
import javafx.scene.control.TabPane;

import java.util.function.Supplier;

public class TabSelect {

    private final TabPane root;
    private Supplier<Tab> notFind;

    private TabSelect(TabPane root) {
        this.root = root;
    }

    public static TabSelect from(TabPane root) {
        return new TabSelect(root);
    }

    public static TabSelect from() {
        return new TabSelect(HomeController.$this.mainTabPanel);
    }

    public TabSelect ifNotFind(Supplier<Tab> notFind) {
        this.notFind = notFind;

        return this;
    }

    public void select(String title) {
        root.getTabs().stream()
                .filter(targetTab -> targetTab.getText().equals(title))
                .findAny()
                .ifPresentOrElse(tab -> root.getSelectionModel().select(tab),
                        () -> HomeController.$this.insertTab(notFind.get()));
    }
}

File: gui_view/src/main/scala/com/goexp/galgame/gui/util/TabSelect.scala
package com.goexp.galgame.gui.util

import com.goexp.galgame.gui.view.game.HomeController
import javafx.scene.control.{Tab, TabPane}

import scala.jdk.CollectionConverters._

object TabSelect {
  def apply(root: TabPane): TabSelect = new TabSelect(root)

  def apply(): TabSelect = new TabSelect(HomeController.$this.mainTabPanel)
}

class TabSelect private(val root: TabPane) {
  private var notFind: () => Tab = _

  def ifNotFind(notFind: () => Tab): TabSelect = {
    this.notFind = notFind
    this
  }

  def select(title: String): Unit =
    root.getTabs.asScala.to(LazyList)
      .find(_.getText == title) match {
      case Some(tab) =>
        root.getSelectionModel.select(tab)
      case None =>
        HomeController.$this.insertTab(notFind())
    }

}